acceptance/analyzer_test.go

```diff
@@ -529,39 +533,31 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
 					}
 				})
 			})
-		})
-
-		when("CNB_REGISTRY_AUTH is not provided", func() {
-			when("DOCKER_CONFIG is set", func() {
-				it("succeeds", func() {
-					h.SkipIf(t, runtime.GOOS == "windows", "Can't run bash on Windows")
-
-					// Copy docker config directory to a temp directory to avoid tampering with permissions for other tests.
-					dockerConfig, err := ioutil.TempDir("", "test-docker-config-")
-					defer os.RemoveAll(dockerConfig)
-					h.AssertNil(t, err)
-					h.RecursiveCopy(t, registry.DockerDirectory, dockerConfig)
-					// Some OSes (like macOS) use symlinks for the standard temp dir.
-					// Resolve it so it can be properly mounted by the Docker daemon.
-					dockerConfig, err = filepath.EvalSymlinks(dockerConfig)
-					h.AssertNil(t, err)
 
-					h.DockerRunAndCopy(t,
-						containerName,
-						copyDir,
-						analyzeImage,
-						"/layers",
-						h.WithFlags(
-							"--mount", fmt.Sprintf("type=bind,source=%s,target=/mounted-docker-config", dockerConfig),
-							"--env", "DOCKER_CONFIG=/mounted-docker-config",
-							"--network", "host",
-						),
-						h.WithBash(
-							fmt.Sprintf("chown -R 2222:3333 /mounted-docker-config; %s %s; ls -alR /layers", analyzerPath, appImage), // provide a real app image, so that we can test that the registry is accessible
-						),
-					)
+			when("CNB_REGISTRY_AUTH is not provided", func() {
+				when("DOCKER_CONFIG is set", func() {
+					it("succeeds", func() {
+						h.DockerRunAndCopy(t,
+							containerName,
+							copyDir,
+							analyzeImage,
+							"/layers",
+							h.WithFlags(
+								"--env", "DOCKER_CONFIG=/registry-docker-config",
+								"--network", "host",
+							),
+							h.WithBash(
+								// ensure docker config directory is root owned and NOT world readable
+								fmt.Sprintf(
+									"chown -R root /registry-docker-config; chmod -R 700 /registry-docker-config; %s %s",
+									analyzerPath,
+									appImage,
+								),
+							),
+						)
 
-					h.AssertPathExists(t, filepath.Join(copyDir, "layers", "some-buildpack-id"))
+						h.AssertPathExists(t, filepath.Join(copyDir, "layers", "some-buildpack-id")) // the presence of some-buildpack-id indicates that we were able to retrieve the app image from the registry
```

Comment:
This test might be more intelligible if we asserted on the log output, e.g., it could assert that the output contains something like `Restoring metadata for "some-buildpack-id:launch-layer" from app image`.

Reply:
Can we do both? Also, is there a reason we check for specific layer file names when using env auth but not when using docker config auth? (e.g. `launch-layer.toml`)

Reply:
Nope, we can make it consistent.

Reply:
Actually on reviewing this again we think that for tests that are only verifying that we successfully connected to the registry (either for the app or the cache) it isn't important to check the specific files. For tests that are verifying the usage of the registry image, it is important. We can make things consistent with this philosophy, or we can just check specific files everywhere.

Reply:
IMO seems worth checking at least one launch layer metadata file to verify that we connected to the registry. Practically, you are right that this test will ensure that our analyzer connects successfully, but reading through the test I found it momentarily confusing that nothing being asserted upon actually required data from the image.

Given that all of the buildpack IDs are provided in the `group.toml` an implementation of `analyzer` could hypothetically decided to create `<layers>/<buildpack ID>` directories even when there is no app image.

Reply:
We're going with consistency :)